### ISSUE TITLE
Images loaded with coil don't show when using transformations

### ISSUE THREAD

**Issue by ubuntudroid** [noties/Markwon#272]
* **Markwon version**: 4.4.0

When using the coil plugin and applying transformations as explained in the [documentation](https://noties.io/Markwon/docs/v4/image-coil/) images are not displayed (only the placeholder appears) when returning to a previously displayed TextView with markdown.

It doesn't seem to matter which transformations you use and how many. Removing the transformations makes the issue go away.

*Markwon configuration:*

```kotlin
Markwon.builder(activity)
    .usePlugin(
        CoilImagesPlugin.create(
            object : CoilImagesPlugin.CoilStore {
                override fun load(drawable: AsyncDrawable): LoadRequest {
                    return LoadRequest.Builder(context)
                        .transformations(
                            RoundedCornersWithoutCropTransformation(14f)
                        )
                        .data(drawable.destination)
                        .placeholder(R.drawable.image_placeholder)
                        .build()
                }

                override fun cancel(disposable: RequestDisposable) {
                    disposable.dispose()
                }   
            },
            ImageLoader.Builder(activity)
                .okHttpClient(okHttpClient)
                .placeholder(R.drawable.image_placeholder)
                .build()
        )
    )
    .build()
```

**Steps to reproduce:**
1. Open a fragment with a TextView that is populated using Markwon with some markdown containing images.
2. Put the app to background.
3. Reopen the app from recents.
4. Observe: the images are not displayed anymore (only placeholders are shown). If an image didn't load in step 2 (e.g. you advanced too quickly or the loading failed) it will load now (if not blocked by some external factors), but also just once.

**Comment by ubuntudroid** [comment]
I've updated the reproduction steps from above: putting the app to background and back already triggers the malfunction. You can actually see how the images are replaced with placeholders when moving the app back to front. 🤔

**Comment by ubuntudroid** [comment]
Here's a GIF of the behaviour:

![CleanShot 2020-07-29 at 18 48 23](https://user-images.githubusercontent.com/519867/88828952-3ae76c00-d1cc-11ea-8099-800887190aec.gif)

**Comment by ubuntudroid** [comment]
Update: it also happens when using the recycler plugin.

**Comment by noties** [comment]
Hello @ubuntudroid ,

it might be connected with the #241 issue which is fixed in latest [snapshot](https://noties.io/Markwon/docs/v4/install.html#snapshot) version. Can you please try your input with it and see if the issue persists?

**Comment by ubuntudroid** [comment]
The snapshot unfortunately doesn't fix the issue. :(

**Comment by noties** [comment]
Hello @ubuntudroid ,

unfortunately I cannot reproduce this issue - [code used](https://noties.io/Markwon/app/sample/20200803132053)

**Comment by ubuntudroid** [comment]
Huh, that's interesting. 🤔 Let me compare my code with yours and get back to you later today.

**Comment by ubuntudroid** [comment]
@noties can you try to not recreate the image loader, but instead use a singleton instance throughout the sample app? For me that seems to be the culprit. But it's obviously no solution to recreate the image loader on every screen as the instances won't share their memory and bitmap cache then.

**Comment by noties** [comment]
Would you @ubuntudroid mind trying to recreate this issue in a sample form? So it is reproducible on my side? You can use [this code](https://github.com/noties/Markwon/blob/master/app-sample/src/main/java/io/noties/markwon/app/samples/image/CoilRecyclerViewSample.kt) as the base

**Comment by ubuntudroid** [comment]
Sure, on it.

**Comment by ubuntudroid** [comment]
@noties here you go. ☝️

**Comment by ubuntudroid** [comment]
Do you need any additional information from me here?

**Comment by noties** [comment]
Hello @ubuntudroid ,

thank for provided info, it helped a lot to deduct the issue. It turns to be a bug in `image-coil` artifact. The first image request goes as usual. And the second request, if image is cached in memory, can return result _before_ request disposable is created. Coil image loader checks if there is a disposable available to understand if request was cancelled. If it does not see a disposable it won't deliver result. This is going to be fixed in next version

**Comment by ubuntudroid** [comment]
@noties That's awesome, many thanks! 🙏 Any rough idea when the version with the fix will be released?

**Comment by noties** [comment]
`4.5.1-SNAPSHOT` with the fix is available right now 😉 . Can you please check if it works for you? If it does a minor stable release can be done in the nearest future

**Comment by ubuntudroid** [comment]
Works perfectly fine! 👏

**Comment by noties** [comment]
Fixed in `4.5.1`
